### DISCUSSION TITLE
Read Energy values in Python?

### DISCUSSION THREAD

**Discussion by 0xCoto** [thliebig/openEMS-Project#74] [Q&A Usage]
Is it somehow possible to pick up the `Energy` values as the simulation is running in Python? I've currently got a separate script to plot the results (time signals, S-Parameters etc.) as the simulation is running in real-time, but I'm not sure how I could pick up the Energy values displayed by openEMS. One solution might be to use string manipulation on openEMS's `stdout`, but I was wondering if there was a nicer/more intended way, similar to e.g. how we're reading the time signals etc. from the `port` Python object.

**Comment by thliebig** [comment]
you can run openEMS with "--dump-statistics" and I think it then writes files with all this infos in it. But I'm not sure how often (or if at all) it flushes these during the simulation.
What do you need this for? Please be aware that it is only a quick and simplified "energy estimation"...

**Reply by thliebig** [reply]
I just noticed that you talked about the python interface. I'm not sure if this is currently available there...
But the best way (for the future) would be to allow a hook to be called with this infos during the simulation run...

**Reply by 0xCoto** [reply]
> What do you need this for? Please be aware that it is only a quick and simplified "energy estimation"...

Yeah, that's fine - I basically want to stream the simulation progress (a percentage estimate, e.g. `energy_dB/end_criteria_dB`) to some external interface, so it would be useful to be able to pick up the current energy as the simulation is running. Perhaps a rough estimate may also be derivable from the port time signals though, since the two quantities tend to be more or less related in some sense (at least in this context)...

**Edit:** Figured it by picking up the stdout, thanks
